### PR TITLE
Modify for CVS-69023(python version): hint configuration

### PR DESCRIPTION
**Pull request by zhaixuejun1993**
Signed-off-by: xuejun <xuejun.zhai@intel.com>

### Details:
finished the python version of https://github.com/openvinotoolkit/openvino/pull/10259

### Tickets:
https://jira.devtools.intel.com/browse/CVS-69023

### CODE REVIEW

**Review by myshevts — APPROVED** [review]

File: tools/benchmark_tool/openvino/tools/benchmark/main.py [hunk 1]
```diff
@@ -80,13 +80,25 @@ def is_flag_set_in_command_line(flag):
...
-                if 'PERFORMANCE_HINT' in supported_properties:
-                    logger.warning(f"-hint default value is determined as 'THROUGHPUT' automatically for {device} device" +
-                                    "For more detailed information look at README.")
+        for device in devices:
+            supported_properties = benchmark.core.get_property(device, 'SUPPORTED_PROPERTIES')
+            if 'PERFORMANCE_HINT' in supported_properties:
+                if is_flag_set_in_command_line('hint'):
+                    if args.perf_hint=='throughput':
+                        logger.warning(f"Device {device} performance hint is set to THROUGHPUT.")
+                        args.perf_hint = "throughput"
```
Comment (nkogteva) [inline tools/benchmark_tool/openvino/tools/benchmark/main.py:89]: Why lower case is used? Looks unaligned with UNDEFINED

File: src/inference/include/ie/ie_plugin_config.hpp [hunk 2]
```diff
@@ -268,6 +268,8 @@ DECLARE_CONFIG_VALUE(MODEL_PRIORITY_LOW);
 DECLARE_CONFIG_KEY(PERFORMANCE_HINT);
 DECLARE_CONFIG_VALUE(LATENCY);
 DECLARE_CONFIG_VALUE(THROUGHPUT);
+DECLARE_CONFIG_VALUE(UNDEFINED);
```
Comment (ilya-lavrenov) [inline src/inference/include/ie/ie_plugin_config.hpp:271]: do other plugins accept undefined value?

File: tools/benchmark_tool/openvino/tools/benchmark/main.py (hunk 1 above)
Comment (zhaixuejun1993) [reply tools/benchmark_tool/openvino/tools/benchmark/main.py:89]: removed UNDEFINED.

File: src/inference/include/ie/ie_plugin_config.hpp (hunk 2 above)
Comment (zhaixuejun1993) [reply src/inference/include/ie/ie_plugin_config.hpp]: removed undefined value

**Comment by myshevts** [comment]
native C++ benchamrk_app now suppports none as a cmd-line value for the hint and throughput as a default value, let's finalize this PR so both tools are aligned.
@nkogteva  @ilya-lavrenov  and @akuporos, were your comments addressed.

Also @zhaixuejun1993, have you tested on the CPU, GPU, AUTO with all possible hints values (you may locally rebase to the latest master without a push)?

File: tools/benchmark_tool/openvino/tools/benchmark/main.py (hunk 1 above)
Comment (myshevts) [reply tools/benchmark_tool/openvino/tools/benchmark/main.py:89]: this assignment looks redundant (already checked with the ==)

File: tools/benchmark_tool/openvino/tools/benchmark/main.py [hunk 3]
```diff
@@ -80,13 +80,25 @@ def is_flag_set_in_command_line(flag):
...
+                    elif args.perf_hint=='latency':
+                        logger.warning(f"Device {device} performance hint is set to LATENCY.")
+                        args.perf_hint = "latency"
```
[the remark this answers is not in the document]
Comment (zhaixuejun1993) [reply tools/benchmark_tool/openvino/tools/benchmark/main.py]: had tested on CPU/GPU/AUTO @myshevts

**Review by wangleis — APPROVED** [review]

**Pull request merged**